synthesizer/src/vm/verify.rs

```diff
@@ -137,40 +137,43 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
         // Get the consensus version.
         let consensus_version = N::CONSENSUS_VERSION(self.block_store().current_block_height())?;
 
-        // Acquire a read lock on the process.
-        let process = self.process.read();
+        // Construct the transaction checksum.
+        let checksum = Data::<Transaction<N>>::Buffer(transaction.to_bytes_le()?.into()).to_checksum::<N>()?;
 
-        // Get the program editions.
-        // If the transaction is an execution
-        //   AND the consensus version is V8 or greater
-        //   AND any of the component program editions (except for `credits.aleo`) are zero
-        // then fail.
-        if let Transaction::Execute(id, _, execution, _) = transaction {
-            if consensus_version >= ConsensusVersion::V8 {
-                for transition in execution.transitions() {
-                    // Get the stack.
-                    let stack = process.get_stack(transition.program_id())?;
-                    // Get the program ID.
-                    let program_id = *stack.program_id();
-                    // If the program edition is zero, then fail.
-                    if program_id != ProgramID::from_str("credits.aleo")? && stack.program_edition().is_zero() {
-                        drop(process); // Drop the process lock. 
-                        bail!(
-                            "Invalid execution transaction '{id}' - the program edition for '{program_id}' cannot be zero for `ConsensusVersion::V8` or greater. Please redeploy the program."
-                        );
-                    }
-                }
+        // Get the program editions from the transaction.
+
+        let mut program_editions = Vec::with_capacity(Transaction::<N>::MAX_TRANSITIONS);
```

Comment:
this capacity would currently be `32`, right? that should be fine, but if we foresee it growing a lot more in the future, this could result in overallocating

Reply:
That's right, also unlikely to grow by too much.
Even a doubling would be a lot.